builders/build/substrate/adding-custom-made-module.md

```diff
@@ -0,0 +1,338 @@
+
+    }
+}
+```
+
+These calls also emit events, to keep the user informed and can return errors, should any of the validations go wrong.
+
+Here is the complete implementation of the calls, with the custom lottery logic:
+
+??? code "View the complete calls code"
+
+    ```rust
+    --8<-- 'code/basic-substrate/lottery-example-calls.rs'
+    ```
+
+### Implementing Custom Errors {: #implementing-custom-errors}
+
+The `#[pallet::error]` macro is applied to an enumeration of errors that might occur during the execution. It is important for security reasons to handle all error cases gracefully and never crash in the runtime.
+
+```rust
+// Errors inform users that something went wrong.
+#[pallet::error]
+pub enum Error<T> {
+    NotEnoughCurrency,
+    AccountAlreadyParticipating,
+    CanNotAddParticipant,
+}
+```
+
+### Implementing Events {: #implementing-events }
+
+The `#[pallet::event]` macro is applied to an enumeration of events to inform the user of any changes in the state or important actions that happened during the execution in the runtime.
+
+The following snippet shows the events defined in the lottery example:
+
+```rust
+#[pallet::event]
+#[pallet::generate_deposit(pub(super) fn deposit_event)]
+pub enum Event<T: Config> {
+    /// Event emitted when a ticket is bought
+    TicketBought { who: T::AccountId },
+    /// Event emitted when the prize is awarded
+    PrizeAwarded { winner: T::AccountId },
+    /// Event emitted when there are no participants
+    ThereAreNoParticipants,
+}
+```
+
+### Implementing Storage for State Persistence {: #implementing-storage }
+
+The `#[pallet::storage]` macro initializes a runtime storage structure. In this example, a basic value storage structure is used to persist the list of participants in a bounded capacity vector ([BoundedVec](https://crates.parity.io/frame_support/storage/bounded_vec/struct.BoundedVec.html){target=_blank}).
+
+In the heavily constrained environment of an AppChain, deciding what to store and which structure to use can be critical in terms of performance. More on this topic is covered in the [Substrate documentation](https://docs.substrate.io/build/runtime-storage/){target=_blank}.
+
+
+```rust
+#[pallet::storage]
+#[pallet::getter(fn get_participants)]
+pub(super) type Participants<T: Config> = StorageValue<
+    _,
+    BoundedVec<T::AccountId, T::MaxParticipants>,
+    OptionQuery
+>;
+```
+### The Complete Module {: #complete-module }
+
+Putting all the pieces together, after implementing all the required macros and adding the custom logic, now the module is complete and ready to be used in the runtime. 
+
+??? code "View the complete module file"
+
+    ```rust
+    --8<-- 'code/modules/lottery-example.rs'
+    ```
+
+## Configure the Runtime {: #configure-runtime }
+
+Finally, with the module finished, it can be included in the runtime. By doing so, the transactions buy_tickets and award_prize will be callable by the users.
```

Comment:
```suggestion
Finally, with the module finished, it can be included in the runtime. By doing so, the transactions `buy_tickets` and `award_prize` will be callable by the users. This also means that the (Polkadot.js API)(/builders/interact/substrate-api/polkadot-js-api/) will be decorated with this module module and all the available calls that it contains.
```

Reply:
(accepted & corrected "module module")